### PR TITLE
Flush signal tests

### PR DESCRIPTION
**Pull request by mmichal10**
Requires https://github.com/intel-innersource/frameworks.validation.opencas.test-framework/pull/36

### CODE REVIEW

File: test/functional/tests/misc/test_flush_propagation.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+mount_point = "/mnt/cas"
+
+
+@pytest.mark.os_dependent
+@pytest.mark.require_plugin("scsi_debug_fua_signals", dev_size_mb="8192", opts="1")
+@pytest.mark.require_disk("core", DiskTypeSet([DiskType.optane, DiskType.nand]))
+def test_flush_signal_propagation_cache():
+    """
+    title: Test for FLUSH signals propagation to cache devuce
```
Comment (pdebski21) [inline test/functional/tests/misc/test_flush_propagation.py:32]: "cache devuce" -> "cache device"

File: test/functional/tests/misc/test_flush_propagation.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Mount exported object."):
+        if core.is_mounted():
+            core.unmount()
+        core.mount(mount_point)
+
+    with TestRun.step("Create temporary file on the exported object."):
+        tmp_file = create_random_test_file(f"{mount_point}/tmp.file", Size(1, Unit.GibiByte))
+        TestRun.executor.run("dmesg -C")
```
Comment (arutk) [inline test/functional/tests/misc/test_flush_propagation.py:107]: what is the purpose of cleaning logs here?

**Review by arutk — CHANGES_REQUESTED** [review]

File: test/functional/tests/misc/test_flush_propagation.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    with TestRun.step(f"Check {syslog_path} for flush request and delete temporary file."):
+        Logs.check_syslog_for_signals()
+        tmp_file.remove(True)
+
+
+@pytest.mark.os_dependent
+@pytest.mark.require_plugin("scsi_debug_fua_signals", dev_size_mb="8192", opts="1")
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+def test_flush_signal_propagation_cache():
```
Comment (pdebski21) [inline test/functional/tests/misc/test_flush_propagation.py:74]: 2 tests have the same name `test_flush_signal_propagation_cache` the difference is required disk so unifying it or add sufix to test name

File: test/functional/tests/misc/test_flush_propagation.py (hunk 2 above)
Comment (mmichal10) [reply test/functional/tests/misc/test_flush_propagation.py:107]: Scsi debug module gathers info about flush request from dmesg. I've added cleaning dmesg to ensure that the flush requests where issued during the actual flush of data to the disk, not i.e. during creating partitions or fs

File: test/functional/tests/misc/test_flush_propagation.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        Logs._read_syslog(Logs.last_read_line)
+
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.scsi_debug_devices[0]
+        core_dev = TestRun.disks["core"]
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start cache on SCSI device and add core with xfs filesystem"):
+        cache = casadm.start_cache(cache_dev, CacheMode.WT)
```
Comment (rafalste) [inline test/functional/tests/misc/test_flush_propagation.py:48]: Shouldn't the cache be in a WB mode?

File: test/functional/tests/misc/test_flush_propagation.py (hunk 2 above)
Comment (rafalste) [reply test/functional/tests/misc/test_flush_propagation.py:107]: Can't you just do it with `Logs._read_syslog(Logs.last_read_line)` like a few lines above?

Comment (mmichal10) [reply test/functional/tests/misc/test_flush_propagation.py]: Done

File: test/functional/tests/misc/test_flush_propagation.py (hunk 4 above)
Comment (mmichal10) [reply test/functional/tests/misc/test_flush_propagation.py:48]: As we expect for the flush signal issued by the OS (not flush of dirty data from cache to core), the cache instance may be running in WT as well

**Review by rafalste — APPROVED** [review]

**Review by pdebski21 — APPROVED** [review]

**Review by arutk — APPROVED** [review]

**Review by pdebski21 — APPROVED** [review]

**Pull request merged**